src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -299,25 +299,27 @@ public UserDetails retrieveUser(String username, String password, String domainN
             }
             LOGGER.fine("Found user "+username+" : "+user);
 
-            Object dn = user.get("distinguishedName").get();
-            if (dn==null)
+            Object userDistinguishedNameObject = user.get("distinguishedName").get();
+            if (userDistinguishedNameObject==null)
                 throw new AuthenticationServiceException("No distinguished name for "+username);
 
+            String userDistinguishedName = escapeCharactersInDistinguisedName(userDistinguishedNameObject.toString(), false);
+
             if (bindName!=null && password!=NO_AUTHENTICATION) {
                 // if we've used the credential specifically for the bind, we
                 // need to verify the provided password to do authentication
-                LOGGER.fine("Attempting to validate password for DN="+dn);
-                DirContext test = descriptor.bind(dn.toString(), password, ldapServers);
+                LOGGER.fine("Attempting to validate password for DN=" + userDistinguishedName);
+                DirContext test = descriptor.bind(userDistinguishedName, password, ldapServers);
                 // Binding alone is not enough to test the credential. Need to actually perform some query operation.
                 // but if the authentication fails this throws an exception
                 try {
-                    new LDAPSearchBuilder(test,domainDN).searchOne("(& (userPrincipalName={0})(objectCategory=user))",userPrincipalName);
+                    new LDAPSearchBuilder(test,domainDN).searchOne("(& (userPrincipalName={0})(objectCategory=user))", userPrincipalName);
                 } finally {
                     closeQuietly(test);
                 }
             }
-
-            Set<GrantedAuthority> groups = resolveGroups(domainDN, dn.toString(), context);
+            String userDistinguishedNameEscapingForwardSlash = escapeCharactersInDistinguisedName(userDistinguishedName, true);
```

Comment:
this escapes more than the forwardSlash.
I would also recommend that this is done inside resolveGroups and not here.